_rules/document-has-headings-for-main-section-of-content-047fe0.md

```diff
@@ -0,0 +1,444 @@
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This [document][] has a `h1` heading at the start of its [main block of content][].
+
+```html
+<html>
+	<head>
+		<title>The Three Kingdoms, Chapter 1</title>
+	</head>
+	<body>
+		<nav id="chapters-navigation">
+			<ol>
+				<li>Chapter 1</li>
+				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
+			</ol>
+		</nav>
+
+		<main>
+			<h1>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h1>
+			<p>
+				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
+				of time.
+			</p>
+		</main>
+	</body>
+</html>
+```
+
+#### Passed Example 2
+
+This [document][] has a `h2` heading at the start of its [main block of content][].
+
+```html
+<html>
+	<head>
+		<title>The Three Kingdoms, Chapter 1</title>
+	</head>
+	<body>
+		<h1>The Three Kingdoms, Chapter 1</h1>
+
+		<nav id="chapters-navigation">
+			<h2>Content</h2>
+			<ol>
+				<li>Chapter 1</li>
+				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
+			</ol>
+		</nav>
+
+		<main>
+			<h2>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h2>
+			<p>
+				Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span
+				of time.
+			</p>
+		</main>
+	</body>
+</html>
+```
+
+#### Passed Example 3
+
+This [document][] has one [block of content][] for the navigation links, and the [main block of content][]. Each starts with a `h1` heading.
+
+**Note:** In this [document][], the [blocks of content][block of content] are identified by the level 1 heading at their start.
+
+```html
+<html>
+	<head>
+		<title>The Three Kingdoms, Chapter 1</title>
+	</head>
+	<body>
+		<!-- Navigational section of content starts here -->
+		<h1>Content</h1>
+		<ol>
+			<li>Chapter 1</li>
+			<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
+		</ol>
+		<!-- Navigational section of content ends here -->
+
+		<!-- Main section of content starts here -->
+		<h1>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</h1>
+		<p>
+			Unity succeeds division and division follows unity. One is bound to be replaced by the other after a long span of
+			time.
+		</p>
+		<!-- Main section of content ends here -->
+	</body>
+</html>
+```
+
+#### Passed Example 4
+
+In this [document][], the heading is not the first element of the [main block of content][], but it is the first with a non-empty [accessible name][] (inherited from its content).
+
+```html
+<html>
+	<head>
+		<title>The Three Kingdoms, Chapter 1</title>
+	</head>
+	<body>
+		<nav id="chapters-navigation">
+			<ol>
+				<li>Chapter 1</li>
+				<li><a href="/test-assets/bypass-blocks-cf77f2/chapter2.html">Chapter 2</a></li>
+			</ol>
+		</nav>
+
+		<main>
+			<hr />
+			<h1><span>Three Heroes Swear Brotherhood at a Feast in the Peach Garden</span></h1>
```

Comment:
The `span` is not needed here.

Reply:
🤔 I wanted to illustrate that it is possible to have some markup inside the heading. I guess a more realistic example would be in the line of `<h1>Lorem <a href="foo">foo</a> Ipsum</h1>`.
But I guess maybe it was indeed too much and not really needed… Removing it.